notebooks/20220328_correlations_mrnorm.ipynb

```diff
@@ -0,0 +1,518 @@
+      "text/plain": [
+       "plot without title"
+      ]
+     },
+     "metadata": {
+      "image/png": {
+       "height": 420,
+       "width": 900
+      }
+     },
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "# add operon metadata and plot\n",
+    "mrnorm_pa14_kmers_cor_df <- cor_to_long(mrnorm_pa14_kmers_cor) %>%\n",
+    "  left_join(pa14_hash_map_operon_small, by = c(\"row\" = \"minhash\")) %>% # determine which operon, if any, the row is part of\n",
+    "  select(row, col, corr, operon_id_row = operon_id) %>% # record operon identity of row\n",
+    "  left_join(pa14_hash_map_operon_small, by = c(\"col\" = \"minhash\")) %>% # determine which operon the col is part of\n",
+    "  select(row, col, corr, operon_id_row, operon_id_col = operon_id) %>% # record operon identity of col\n",
+    "  mutate(operon = ifelse(operon_id_row == operon_id_col, \"operon pair\", \"not operon pair\")) %>% # label operon pairs\n",
+    "  mutate(operon = ifelse(is.na(operon), \"not operon pair\", operon))\n",
+    "\n",
+    "ggplot(mrnorm_pa14_kmers_cor_df, aes(x = corr, fill = operon)) +\n",
+    "  geom_histogram(alpha = .5) +\n",
+    "  theme_classic()"
+   ]
```

Comment:
ok, just so I make sure I understand this correctly. You are calculating the correlation between each kmer, then coloring it by whether the k-mers are within an operon-operon across operons?

Then since the k-mers are supposed to be unique to an operon, you would expect the counts of k-mers from the same operon to be highly correlated?

Reply:
> its interesting that its mostly positive correlation, is this expected? Sorry, I don't quite follow what this plot should show.

I don't have strong notions about what that this should look like, other than expecting it to look very similar to gene count correlations. 

> also, maybe useful to try spearman correlation too, in case this is being driven by a couple outlier samples

ooh good idea. i'll give this a try

> ok, just so I make sure I understand this correctly. You are calculating the correlation between each kmer, then coloring it by whether the k-mers are within an operon-operon across operons?
> Then since the k-mers are supposed to be unique to an operon, you would expect the counts of k-mers from the same operon to be highly correlated?

Yes this is exactly right.